File: PHP/src/Jakopec/PartijaTriIgraca.php
<?php


namespace Jakopec;


class PartijaTriIgraca extends Partija
{

    public function __toString(): string
    {
        $rezultat = $this->getRezultat();
        return 'Partija TRI IGRAČA, igra gotova: ' . $this->isIgraGotova() . ', ' . $this->getIgraci()[0] . ' i ' . ': ' .
            $rezultat->getPrvi() .
            ' | ' . $this->getIgraci()[1] . ': ' . $rezultat->getDrugi() .
            ' | ' . $this->getIgraci()[2] . ': ' . $rezultat->getTreci();
    }

    public function getRezultat(): Rezultat
    {
        $rezultat = parent::getRezultat();

        $b = 0;
        foreach ($this->getMjesanja() as $mjesanje) {
            $b += $mjesanje->getRezultat()->getTreci();
        }
        $rezultat->setTreci($b);

        return $rezultat;
    }
}

File: Java/src/jakopec/model/PartijaTriIgraca.java
package jakopec.model;

public class PartijaTriIgraca extends PartijaDvaIgraca {

    @Override
    public String toString() {
        Rezulat rezulat = getRezultat();
        return "Partija TRI IGRAČA, igra gotova: " + isIgraGotova() + ", " +
                getIgraci().get(0) + ": " + rezulat.getPrvi() +" | "+
                getIgraci().get(1) + ": " + rezulat.getDrugi() + " | "+
                getIgraci().get(2) + ": " + rezulat.getTreci();
    }

    @Override
    public Rezulat getRezultat() {
        Rezulat rezultat = super.getRezultat();
                rezultat.setTreci(getMjesanja().stream().mapToInt(x -> x.getRezultat().getTreci()).sum());
        return rezultat;
    }
}
